### DISCUSSION TITLE
Generate a video preview with user's choice

### DISCUSSION THREAD

**Discussion by tiberio87** [Audionut/Upload-Assistant#139] [Ideas]
Using this command from terminal with ffmpeg you get a video as a sample, ffmpeg -i Title.Title.2024.1080p.BluRay.DD.5.1.H.265-GROUP.mkv -ss 00:30:00 -t 00:01:00 -c copy Title.Title.2024.1080p.BluRay.DD.5.1.H.265-GROUP.SAMPLE.mkv

You can implement this work by choosing each time if you want to generate the sample of the video, if you choose it then the script must load a folder with the full movie and the sample, if instead you do not want to generate the sample the script will load without a folder.

**Comment by Audionut** [comment]
Fluff, such as sample files are the bane of existence for cross-seeding. Sites who allow sample files and so forth, generally end with content that is both dead, and cannot be easily reseeded. I won't support implementing this in the main code.

I would accept a PR for a specific site file though. Look at MTV for ideas https://github.com/Audionut/Upload-Assistant/blob/b9f8f1a296184d9a2087bc33c205f35b1be41910/src/trackers/MTV.py#L109

You'll need to adjust the condition that causes the torrent to be recreated, adjust the include/exclude globs and probably set the keep-folder meta (and probably set back to false when finished). You should be able to have a file in a folder without --keep-folder for other sites, and then adjust to suit.

Create a function inside of the site file for ffmpeg processing.
